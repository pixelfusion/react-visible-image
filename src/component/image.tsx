import * as React from 'react'
import { Base, BaseProps, propsToStrip } from '../base'
import { cleanProps } from '../utility/clean-props'

export interface ImageProps extends BaseProps {
    alt?: any
    height?: any
    width?: any
    poster?: any
}

export class Image extends Base<ImageProps> {
    render(): JSX.Element {
        const { initialImage, image } = this.props
        const { isVisible, hasIntersectionObserver } = this.state
        const imageSrc: string = (isVisible && hasIntersectionObserver) ? image : initialImage
        const attributes: object = cleanProps(this.props, propsToStrip)

        return (
            <img
                {...attributes}
                className={this.getClassName()}
                src={imageSrc}
                ref={(element) => { this.target = element }}
            />
        )
    }
}
