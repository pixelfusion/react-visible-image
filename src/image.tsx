import * as React from 'react'
import { Base, BaseProps, propsToStrip } from './base'
import cleanProperties from './utility/clean-props'

export interface ImageProps extends BaseProps {}

export class Image extends Base<ImageProps> {
    render(): JSX.Element {
        const { initialImage, image } = this.props;
        const imageSrc: string = (this.state.isVisible) ? image : initialImage;
        const attributes: object = cleanProperties(this.props, propsToStrip);

        return (
            <img {...attributes} className={this.getClassName()} src={imageSrc} ref={(element) => { this.target = element; }} />
        );
    }
}
