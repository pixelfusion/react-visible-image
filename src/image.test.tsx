import * as React from 'react'
import * as TestUtils from 'react-dom/test-utils'
import * as renderer from 'react-test-renderer'
import { Image } from './image'

test('renders correctly with an image', () => {
    const tree = renderer.create(
        <Image initialImage="https://twitter.com" image="http://www.facebook.com" />
    ).toJSON()

    expect(tree).toMatchSnapshot()
})

test('renders correctly without an image', () => {
    const tree = renderer.create(
        <Image image="http://www.facebook.com" />
    ).toJSON()

    expect(tree).toMatchSnapshot()
})

test('should return the default loading class name', () => {
    const image = TestUtils.renderIntoDocument(<Image image="http://github.com" />)

    expect(image.getClassName()).toBe('is-loading')
})

test('should return a provided loading class name', () => {
    const image = TestUtils.renderIntoDocument(<Image image="http://github.com" loadingClassName="test-class" />)

    expect(image.getClassName()).toBe('test-class')
})

test('should not return a loading class name when image is visible', () => {
    const image = TestUtils.renderIntoDocument(<Image image="http://github.com" shouldShow={true} />)

    expect(image.getClassName()).toBe('')
})

test('should not return a loading class name when image is visible and loading class is passed', () => {
    const image = TestUtils.renderIntoDocument(<Image image="http://github.com" loadingClassName="test-class" shouldShow={true} />)

    expect(image.getClassName()).not.toContain('test-class')
})

test('should return a class name when image is visible and loading class is passed', () => {
    const image = TestUtils.renderIntoDocument(<Image className="my-class" image="http://github.com" loadingClassName="test-class" shouldShow={true} />)

    expect(image.getClassName()).toBe('my-class')
})

test('should return a class name when image is visible and loading class is passed', () => {
    const image = TestUtils.renderIntoDocument(<Image className="my-class" image="http://github.com" loadingClassName="test-class" shouldShow={true} />)

    expect(image.getClassName()).toBe('my-class')
})
