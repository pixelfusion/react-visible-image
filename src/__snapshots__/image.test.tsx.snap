// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with an image 1`] = `
<img
  className="is-loading"
  src="https://twitter.com"
/>
`;

exports[`renders correctly without an image 1`] = `
<img
  className="is-loading"
  src={undefined}
/>
`;
